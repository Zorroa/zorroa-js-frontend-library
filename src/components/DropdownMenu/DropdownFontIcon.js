import React, { PropTypes, PureComponent } from 'react'
import classnames from 'classnames'

class DropdownFontIcon extends PureComponent {
  static propTypes = {
    dark: PropTypes.bool.isRequired,
    children: PropTypes.any,
    icon: PropTypes.string.isRequired,
  }

  render() {
    const dropdownItemClasses = classnames(
      'DropdownMenu__font-icon',
      this.props.icon,
      {
        'DropdownMenu__font-icon--dark': this.props.dark === true,
      },
    )
    return <div className={dropdownItemClasses}>{this.props.children}</div>
  }
}

export default DropdownFontIcon
