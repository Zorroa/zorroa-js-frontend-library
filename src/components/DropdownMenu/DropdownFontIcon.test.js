/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import DropdownFontIcon from './DropdownFontIcon'

describe('<DropdownFontIcon />', () => {
  it('It should render the proper markup', () => {
    const tree = renderer
      .create(<DropdownFontIcon icon="icon-move-left" dark={false} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
