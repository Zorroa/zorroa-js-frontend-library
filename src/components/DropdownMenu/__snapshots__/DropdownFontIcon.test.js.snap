// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownFontIcon /> It should render the proper markup 1`] = `
<div
  className="DropdownMenu__font-icon icon-move-left"
/>
`;
